include/deal.II/lac/petsc_ts.h

```diff
@@ -565,6 +602,61 @@ namespace PETScWrappers
      */
     std::function<IndexSet()> algebraic_components;
 
+    /**
+     * Callback to distribute solution to hanging nodes.
+     *
+     * Implementation of this function is optional.
+     * It is called at the end of each successful stage.
+     * The same functionality can be equivalently implemented in
+     * TimeStepper::solve_with_jacobian.
+     *
+     * @note This variable represents a
+     * @ref GlossUserProvidedCallBack "user provided callback".
+     * See there for a description of how to deal with errors and other
+     * requirements and conventions.
+     */
+    std::function<void(const real_type t, VectorType &y)> distribute;
```

Comment:
For me is not that important since the only thing this method does is to fix the hanging nodes values. Doing it in specific callback is  a convenience: however, what @luca-heltai wrote here in the docs clearly describe why you want to have it.